smartsets.htm

```diff
@@ -1,135 +1,185 @@
-        setbox = '<div class="collapse" id="set'+set.id+'-'+uniqid+'"><div class="card card-body" style="display: inline-block;">Reward: '+set.reward+'<br />';
-        for(row of set.cards) {
-            var card = cardtemplate;
-            card = card.replace(/{ID}/g, row.id.toString());
-            card = card.replace(/{IMAGE}/g, row.image);
-            card = card.replace(/{NAME}/g, row.name);
-            card = card.replace(/{SERIES}/g, row.series);
-            card = card.replace(/{RARITY}/g, row.rarity);
-            if(user != "null") {
-                card = card.replace(/{OWNEDCLASS}/g, row.owned ? " owned" : " not-owned");
-                card = card.replace(/{OWNEDDATA}/g, '<div class="owned-icon">' + (row.owned ? "✔" : "❌") + "</div>");
-            }
-            else {
-                card = card.replace(/{OWNEDCLASS}/g, '').replace(/{OWNEDDATA}/g, '');
+    let uniqid = 0;
+    let cardtemplate = '<div class="card card-tcg">' +
+        '<div class="card-body card-body-tcg">' +
+        '<img data-src="{IMAGE}" alt="{NAME}" title="{NAME}" class="card-image lazyload" />' +
+        '<div class="id-holder rarity-{RARITY}">{ID}</div>' +
+        '<div class="invisible-space-holder">&nbsp;</div>' +
+        '<div class="rarity-holder rarity-{RARITY}">{RARITY}</div>' +
+        '</div>' +
+        '<div class="card-footer text-center{OWNEDCLASS}">' +
+        '{OWNEDDATA}' +
+        '<div class="card-info">' +
+        '{NAME}<br />' +
+        '{SERIES}' +
+        '</div>' +
+        '</div>' +
+        '</div>';
+
+    function populateSets(destination, data, emptyString) {
+        if ('error' in data || !('count' in data && 'sets' in data)) {
+            $(destination).html("<i>Data loading error. Please try again.</i>");
+            return;
+        }
+        if (data.count === 0) {
+            $(destination).html("<i>" + emptyString + "</i>");
+            return;
+        }
+        uniqid += 1;
+        let buttons = "";
+        let content = "";
+        buttonTemplate = '<button class="btn btn-primary btn-set" type="button" data-toggle="collapse" data-target="#set{ID}-{UNIQ}" aria-expanded="false" aria-controls="set{ID}-{UNIQ}">{TITLE}</button>&nbsp;';
+        for (set of data.sets) {
+            title = user !== "null" ? set.name + " (" + set.cardsOwned + "/" + set.totalCards + ")" : set.name;
+            buttons += buttonTemplate.replace(/{ID}/g, set.id).replace(/{UNIQ}/g, uniqid).replace(/{TITLE}/g, title);
+            setbox = '<div class="collapse" id="set' + set.id + '-' + uniqid + '"><div class="card card-body" style="display: inline-block;">Reward: ' + set.reward + '<br />';
+            for (row of set.cards) {
+                let card = cardtemplate;
+                card = card.replace(/{ID}/g, row.id.toString());
+                card = card.replace(/{IMAGE}/g, row.image);
+                card = card.replace(/{NAME}/g, row.name);
```

Comment:
This will interfere with {NAME} being the user's name - replace this one with something like {CARDNAME} (and in the `let cardtemplate =` as well)

Reply:
It won't as the value is replaced inside the loop and this does not touch the {NAME} value in the Navbar.

Reply:
Other way round, this one is in the template and will itself be replaced by the node code.

Reply:
To be fair this code is almost certainly still going to function with the user's name replacing {NAME} in the template's own JS, it's just a weird cosmetic thing and could lead to some unexpected results if people are called something like "div"